services/text-formatters.spec.js

```diff
@@ -44,6 +44,11 @@ describe('Text formatters', function() {
     given(1000).expect('1k')
     given(999499).expect('999k')
     given(999500).expect('1M')
+    given(1049999).expect('1M')
+    given(1050000).expect('1.1M')
+    given(1549999).expect('1.5M')
+    given(10499999).expect('10M')
+    given(10500000).expect('11M')
```

Comment:
What happens if you call e.g. `metric(1234567.89')`? I think what we want is 1.2M though I'm not sure it will currently do the right thing.